assets/src/stories-editor/blocks/amp-story-page/animation-settings.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	PanelBody,
+	IconButton,
+} from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import stopIcon from '../../../../images/stories-editor/stop.svg';
+
+/**
+ * Displays the animation settings for the page block.
+ *
+ * It allows the user to preview all animations on the current page.
+ *
+ * @param {Object} props Component props.
+ * @param {string} props.clientId Client ID.
+ */
+const AnimationSettings = ( { clientId } ) => {
```

Comment:
This component `AnimationSettings ` is almost identical to the `AnimationControls` button. Could we somehow reuse both the logic and the presentation and generalise the few differences?

This will also have the added benefit of rewriting `AnimationControls` to use hooks instead of HoC's.

Reply:
I thought about it, but it seemed like an unnecessary abstraction for such a small component.